### PR TITLE
[Enhancement] Support querying Iceberg tables that have been REPLACE PARTITION FIELD

### PR DESCRIPTION
**Pull request by lipenglin**
Why I'm doing:

```
spark-sql>
CREATE TABLE hive_prod.testdb.sample (id bigint, data string, category string)
USING iceberg PARTITIONED BY (category);
ALTER TABLE hive_prod.testdb.sample REPLACE PARTITION FIELD category with category AS category_111;
INSERT INTO hive_prod.testdb.sample VALUES (1, 'a', 'A'), (2, 'b', 'B'), (3, 'c', 'C');
```

```
spark-sql> select * from hive_prod.testdb.sample;
2       b       B
3       c       C
1       a       A
Time taken: 0.437 seconds, Fetched 3 row(s)
```

```
impala> select * from sample;
+----+------+----------+
| id | data | category |
+----+------+----------+
| 3  | c    | C        |
| 1  | a    | A        |
| 2  | b    | B        |
+----+------+----------+
```

```
sr:
MySQL [testdb]> select * from sample;
ERROR 1064 (HY000): Cannot invoke "com.starrocks.catalog.Column.getName()" because "x" is null
```

What I'm doing:

Iceberg tables `field.name` may not necessarily be `partition-spec.field.name` for `identity Partition Transforms`,  `field.name` should be identified using `partition-spec.field.source-id` in `schema`.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java [hunk 1]
```diff
@@ -793,4 +794,8 @@ public static String getPartitionName(String basePath, String partitionPath) {
     public static String getPathWithSlash(String path) {
         return path.endsWith("/") ? path : path + "/";
     }
+
+    public static String getPartitionSourceName(Schema schema, PartitionField partition) {
+        return schema.findColumnName(partition.sourceId());
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java:799]: Can you comment here to explain the difference from PartitionField.name() and elaborate on when to use which?

Comment (lipenglin) [reply fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java:828]: Thanks for comments!

File: fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java [hunk 2]
```diff
@@ -793,4 +794,37 @@ public static String getPartitionName(String basePath, String partitionPath) {
...
+     * <p>
+     *     To obtain the table columns for Iceberg tables, we use 'org.apache.iceberg.Schema#findColumnName'.
+     * </p>
+     *<br>
+     * refs:<br>
+     * - https://iceberg.apache.org/spec/#partition-evolution<br>
+     * - https://iceberg.apache.org/spec/#partition-specs<br>
+     * - https://iceberg.apache.org/spec/#partition-transforms
+     */
+    public static String getPartitionSourceName(Schema schema, PartitionField partition) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java:827]: You could put this at IcebergPartitionUtils

Comment (lipenglin) [reply fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java]: done

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java [hunk 3]
```diff
@@ -210,6 +210,39 @@ public boolean isV2Format() {
...
+     *     To obtain the table columns for Iceberg tables, we use 'org.apache.iceberg.Schema#findColumnName'.
+     * </p>
+     *<br>
+     * refs:<br>
+     * - https://iceberg.apache.org/spec/#partition-evolution<br>
+     * - https://iceberg.apache.org/spec/#partition-specs<br>
+     * - https://iceberg.apache.org/spec/#partition-transforms
+     */
+    public String getPartitionSourceName(Schema schema, PartitionField partition) {
+        return schema.findColumnName(partition.sourceId());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:243]: LGTM,We may refactor this part after fully supporting the transformed partition.

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**